core/src/Streamly/Unicode/Stream.hs

```diff
@@ -75,18 +75,25 @@
 --
 module Streamly.Unicode.Stream
     (
-    -- * Construction (Decoding)
-      decodeLatin1
+      DecodeState
+    , CodePoint
+    , DecodeError(..)
+
+    -- * Resumable UTF-8 decoding
+    , decodeLatin1
     , decodeUtf8
     , decodeUtf8'
     , decodeUtf8Chunks
+    , decodeUtf8Either
+    , resumeDecodeUtf8Either
```

Comment:
Add documentation for this API and in the documentation mention that the initial DecodeState  is 0 and initialCodePoint is 0.